drivers/sensor/fxos8700/fxos8700.c

```diff
@@ -274,19 +273,32 @@ static int fxos8700_init(struct device *dev)
 		return -EINVAL;
 	}
 
-	/* Read the WHOAMI register to make sure we are talking to FXOS8700 and
-	 * not some other type of device that happens to have the same I2C
-	 * address.
-	*/
+	/*
+	 * Read the WHOAMI register to make sure we are talking to FXOS8700 or
+	 * compatible device and not some other type of device that happens to
+	 * have the same I2C address.
+	 */
 	if (i2c_reg_read_byte(data->i2c, config->i2c_address,
-			      FXOS8700_REG_WHOAMI, &whoami)) {
+			      FXOS8700_REG_WHOAMI, &data->whoami)) {
```

Comment:
Why are you saving this into the data struct? Do you plan to use it again later?

Reply:
I have moved it here from fxos8700_config, it can be used to check if the device supports certain feature (e.g, in fxos8700_attr_set).